lib/action_cable/channel/naming.rb

```diff
@@ -0,0 +1,23 @@
+module ActionCable
+  module Channel
+    module Naming
+      extend ActiveSupport::Concern
+
+      class_methods do
```

Comment:
I tried this but the block was evaluated in `ActionCable::Channel::Base` instead of each of the subclasses so it was not generating the correct name.